feat(ui-kit): coachingSessionTracker component , storybook and test are added.

manishRise11 opened the pull request:
## Description

<!-- Provide a clear and concise description of the changes made in this pull request for UI-Kit components. -->

- Component name: `CoachingSessionsTracker`
- Figma link or screenshot: [Figma Design](https://www.figma.com/design/8KEwRuOoD5IgxTtFAtLlyS/Just_Do_Ad-1.2?node-id=5685-115255&t=2ZDV6HJxfQDBvOSW-4).

![Image](https://github.com/user-attachments/assets/46d688c9-4c2f-4ee2-a3f3-84155bb28cdf)

---


## React Best Practices Checklist

- [x] I have avoided **unnecessary re-renders** by using composition (`children` as props), whenever possible. This applies, in particular, to composite components (like lists of components, and the like).
- [x] **I have designed my components as server-side whenever possible**—I have not used `useState`, `useEffect`, `useContext`, or any other React hook that works with "use client" without a good reason.
- [x] **I have not put props in state**—in general, props should remain as props unless the component needs an `initialValue`, and judiciously manage its own independent state detached from the parent component.
    - If I have stored props in state, I have explained where exactly and justified it with a solid reason here below:
    
- [x] **I have avoided using Context** in any critical component in terms of performance
    - If I have used Context, I have explained where exactly and justified it with a solid reason here below:


## General Checklist

- [x] I have run the lint, build, and test commands locally, as explained in the README, before creating this PR to ensure the code is clean and functional.
- [x] My changes align with the **core models** that have been discussed and agreed upon.
- [x] The component design has been compared to the **Figma design** to ensure consistency.
- [x] The **component is fully responsive** and behaves as expected across different screen sizes.
- [x] The component **supports keyboard web accessibility** (if applicable).
- [x] I have written **unit tests** or **integration tests** to cover the component's functionality above the coverage threshold.
- [x] I have updated the **documentation** for the component, including usage examples and guidelines.
- [x] I have **commented** my code, particularly in complex or non-obvious areas.
- [x] The **component is properly exported** for use in other parts of the application.




---

## Related Issues

<!-- Link any related issue or task. For example: -->
Fixes #154

manishRise11 commented:
Hello @alebg sir, in this CoachingSessionTracker component, I'm passing the session cards as children. Should I use .map() instead, or is this correct?

alebg commented:
> Hello @alebg sir, in this CoachingSessionTracker component, I'm passing the session cards as children. Should I use .map() instead, or is this correct?

Hi @manishRise11 . Thanks a lot, that approach is excellent for this case! No need to map props. We'll compose it in the parent, just as you have shown in the story. Looks very good to me overall, but let me give it a further look, and  let's also wait for @carlapbg 's review

alebg commented on packages/ui-kit/lib/components/coaching-session-tracker/coaching-session-card.tsx:
@manishRise11 please rename this to `coaching-session-item.tsx`, as we're using the name "card" to refer to something else in the project, and it will get confusing

alebg commented on packages/ui-kit/lib/components/coaching-session-tracker/coaching-session-card.tsx:
```diff
@@ -0,0 +1,61 @@
+import { FC } from "react";
+import { Badge } from "../badge";
+
+export interface CoachingSessionCardProps extends TCoachingOffer , isLocalAware {
```
@manishRise11 same, `CoachingSessionItemProps`

alebg commented on packages/ui-kit/lib/components/coaching-session-tracker/coaching-session-tracker.tsx:
```diff
@@ -0,0 +1,58 @@
+                    {dictionary.components.coachingSessionTracker.coachingSessionText}
+                </p>
+                <div className="flex gap-2 flex-wrap">
+                    {children}
```
@manishRise11 nothing to change here, just to say that the children as props approach here works very nicely. Thanks for thinking this one through!!

alebg commented on packages/ui-kit/lib/components/coaching-session-tracker/coaching-session-tracker.tsx:
```diff
@@ -0,0 +1,58 @@
+
+export const CoachingSessionTracker: FC<CoachingSessionTrackerProps> = ({
+    children,
+    onClickByMoreSessions,
```
@manishRise11 This is a typo, should read "Buy" not "By". Please rename all variables strings where this typo appears (this is not the only place)

alebg commented on packages/ui-kit/stories/coaching-session-tracker.stories.tsx:
```diff
@@ -0,0 +1,97 @@
+export const GermanLocale: Story = {
+  render: (args) => (
+    <CoachingSessionTracker {...args} onClickByMoreSessions={() => alert('Buy more coaching sessions')}>
+      {coachingSessions.map((session, index) => (
```
@manishRise11 yes very good, this is how we'll use this in the pages!

alebg reviewed (changes requested):
@manishRise11 overall it looks very good, thanks a lot! Just please address the minor inline comments I made. Also, could you please add a couple more stories, with more `CoachingSessionItems` to see how do they render when we have a lot of them?

In the figma I don't see what happens if they start to pack. @carlapbg do you think implementing something like a horizontal scroller would be a good idea if the items start getting too close to the edge of the container?

carlapbg commented:
> @manishRise11 overall it looks very good, thanks a lot! Just please address the minor inline comments I made. Also, could you please add a couple more stories, with more `CoachingSessionItems` to see how do they render when we have a lot of them?
> 
> In the figma I don't see what happens if they start to pack. @carlapbg do you think implementing something like a horizontal scroller would be a good idea if the items start getting too close to the edge of the container?

If there are going to be more items, then I think it would be a good idea to show a second line (that way it’s easier to see and find everything included in the coaching sessions). I don’t think there will be too many items, but just in case that second line fills up, we can add a vertical scroller.

alebg commented:
> > @manishRise11 overall it looks very good, thanks a lot! Just please address the minor inline comments I made. Also, could you please add a couple more stories, with more `CoachingSessionItems` to see how do they render when we have a lot of them?
> > In the figma I don't see what happens if they start to pack. @carlapbg do you think implementing something like a horizontal scroller would be a good idea if the items start getting too close to the edge of the container?
> 
> If there are going to be more items, then I think it would be a good idea to show a second line (that way it’s easier to see and find everything included in the coaching sessions). I don’t think there will be too many items, but just in case that second line fills up, we can add a vertical scroller.

Right, so more or less like for the skill badges in the coach cards. Perfect then. @manishRise11 could you please implement this? (keeping the children as props strategy, which is very good)

manishRise11 commented:
@carlapbg, ma'am, I have one query regarding the coaching session item. I'm currently using a fixed width of 219px. It looks fine in both 'en' and 'de' as per the attached image. Could you please check and let me know if it’s okay or if I should make any changes?
<img width="1125" alt="Screenshot 2025-04-25 at 7 17 47 PM" src="https://github.com/user-attachments/assets/bddf0b3b-df84-41d8-997f-4be91e3d1f72" />
<img width="1125" alt="Screenshot 2025-04-25 at 7 18 12 PM" src="https://github.com/user-attachments/assets/e7f2f334-5f17-47d5-89ef-5d9e1361e68b" />

manishRise11 commented:
Hi @alebg  sir,
I’ve made the following updates in this commit:


- Renamed CoachingSessionCard to CoachingSessionItem.
- Added story for multiple items.
- Resolved the typo error.
- Enabled horizontal scroll for items.

alebg commented on packages/translations/src/lib/dictionaries/base.ts:
```diff
@@ -211,6 +211,36 @@ export const DictionarySchema = z.object({
+      coachText: z.string(),
+      courseCreatorText: z.string(),
+    }),
+    coachingSessionCard: z.object({
```
hey @manishRise11 , what are all of these?

alebg reviewed (changes requested):
@manishRise11 thanks a lot. Also for adding the empty state story. We didn't think about this case before, and I'm not sure if we will use it at all, but just in case, could you please implement the empty state a bit differently?

So instead of saying 
"Coaching sessions included"
it should say "No coaching sessions included"

and instead of "Buy more sessions", it should say "But coaching sessions"

both translated to German. Apart from that, it looks very good code-wise, but let's wait for @carlapbg 's re-check. In some cases I think the scroll bar is too close to the items though:

![immagine](https://github.com/user-attachments/assets/6d4bad6b-bb2d-43aa-a8f9-db3e652ccebe)

carlapbg commented:
> @carlapbg, ma'am, I have one query regarding the coaching session item. I'm currently using a fixed width of 219px. It looks fine in both 'en' and 'de' as per the attached image. Could you please check and let me know if it’s okay or if I should make any changes? <img alt="Screenshot 2025-04-25 at 7 17 47 PM" width="1125" src="https://private-user-images.githubusercontent.com/188843764/437470109-bddf0b3b-df84-41d8-997f-4be91e3d1f72.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU2ODg4NjgsIm5iZiI6MTc0NTY4ODU2OCwicGF0aCI6Ii8xODg4NDM3NjQvNDM3NDcwMTA5LWJkZGYwYjNiLWRmODQtNDFkOC05OTdmLTRiZTkxZTNkMWY3Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQyNlQxNzI5MjhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNjNhMTRkZGM0MDMzYTg0NmY4YThlMmU0OGNmNmE0M2UyMDExM2RmYzczOGFmZDE0NTUxOTcwMDgyYzY3NTkzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.gWv6e8F4vYpU4P2Zz84rJhpsaNcUcBv9itQf8p-8hIM"> <img alt="Screenshot 2025-04-25 at 7 18 12 PM" width="1125" src="https://private-user-images.githubusercontent.com/188843764/437470374-e7f2f334-5f17-47d5-89ef-5d9e1361e68b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU2ODg4NjgsIm5iZiI6MTc0NTY4ODU2OCwicGF0aCI6Ii8xODg4NDM3NjQvNDM3NDcwMzc0LWU3ZjJmMzM0LTVmMTctNDdkNS04OWVmLTVkOWUxMzYxZTY4Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQyNlQxNzI5MjhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yZmNkNWI0NDBhOGExZDVmZWFjZmFmNGFhMjNkYmVmZTRlZTA5NTMwYmQ1Mjk1NjQxYTEwMDA1YmVhMmNhYTM0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.iII5bHVxiGQBT-bP6ZigZPT7Y1yJ_lmPDGVvsupmMWc">

Maybe let's not used a fixed width. I think for the german side it looks kinda weird because all the words are bigger. Every item can have a width of 230px, so it looks better on the german part.

carlapbg reviewed (changes requested):
Hi! Apart for the comment I left, can you remove the scrollbar when it's not used? For example in the default option we still see it really close to the items. It should only appear when we have more than 2 lines

manishRise11 commented:
@alebg sir and @carlapbg ma’am, I’ve made the following updates:

- Changed the empty state text from “Coaching sessions included” to “No coaching sessions included.”
- Updated the button label in the empty state from “Buy more sessions” to “Buy coaching session.s”
- Added German translations for the new empty state text
- Increased item width from 219px to 230px to better support German text
- Made the scrollbar visible only when content overflows (dynamic visibility)
- Adjusted scrollbar spacing to prevent overlap with items
- Renamed the callback from ‘onClickBuyMoreSessions’ to ‘onClickBuySessions’

alebg reviewed (dismissed):
@manishRise11 Looks good to me, thanks a lot! Let's wait for @carlapbg 's approval and then we're ready to merge

carlapbg reviewed (approved):
Looks good to me, excellent work! Thank you
